### ISSUE TITLE
Update dependencies to latest versions

### ISSUE THREAD

**Issue by chitralverma** [sfu-db/connector-x#493]
#### Describe your feature request

A lot of required and optional dependencies that are mentioned in `connector-x`'s Cargo.toml are due for an update. Listing down some of the important ones,

- Arrow 
- Datafusion
- Polars
- sqlparser

**Comment by chitralverma** [comment]
@wangxiaoying if this makes sense to you, I can raise a PR for this?

**Comment by houqp** [comment]
@chitralverma yes, I have sent updates for datafusion and arrow to connectorx before :)

**Comment by wangxiaoying** [comment]
Hi @chitralverma , thanks for opening this issue and summarizing it. Sure please feel free to open a PR. Just a reminder that other than `connectorx`, you may also want to sync the dependencies in `connectorx-python` as well as `connectorx-cpp`.

**Comment by gruuya** [comment]
If #518 is adequate, can we also have a new release? Even an alpha release will do, since we want to be able to publish the [`datafusion-remote-tables`](https://github.com/splitgraph/seafowl/tree/main/datafusion_remote_tables) crate. For some background see https://github.com/splitgraph/seafowl/issues/456.
